fv3core/stencils/map_single.py

```diff
@@ -188,31 +190,34 @@ def setup_data(
     pe1: FloatField,
     i1: int,
     i2: int,
-    j_2d: Optional[int] = None,
-    j_interface: bool = False,
+    j1: int,
+    j2: int
+    # j_2d: Optional[int] = None,
+    # j_interface: bool = False,
 ):
     grid = spec.grid
     i_extent = i2 - i1 + 1
-    origin, domain, jslice = region_mode(j_2d, i1, i_extent, grid)
-    if j_interface:
-        jslice = slice(jslice.start, jslice.stop + 1)
-        domain = (domain[0], jslice.stop - jslice.start, domain[2])
+    j_extent = j2 - j1 + 1
+    origin, domain = region_mode(i1, i_extent, j1, j_extent, grid)
+    # if j_interface:
+    #     jslice = slice(jslice.start, jslice.stop + 1)
+    #     domain = (domain[0], jslice.stop - jslice.start, domain[2])
 
     dp1 = utils.make_storage_from_shape(
         q1.shape, origin=origin, cache_key="map_single_dp1"
     )
     q4_1 = copy(q1, origin=(0, 0, 0), domain=grid.domain_shape_full())
     q4_2 = utils.make_storage_from_shape(
-        q4_1.shape, origin=(grid.is_, 0, 0), cache_key="map_single_q4_2"
+        q4_1.shape, origin=(grid.is_, grid.js, 0), cache_key="map_single_q4_2"
```

Comment:
could do grid.compute_origin() for here and the other ones in this method, but am not going to insist